packages/expo-updates/android/src/main/java/expo/modules/updates/procedures/RecreateReactContextProcedure.kt

```diff
@@ -1,31 +1,32 @@
 package expo.modules.updates.procedures
 
+import android.app.Activity
+import android.content.Context
 import android.os.Handler
 import android.os.Looper
-import com.facebook.react.ReactNativeHost
+import com.facebook.react.ReactApplication
 import expo.modules.updates.launcher.Launcher
 import expo.modules.updates.statemachine.UpdatesStateEvent
 import java.lang.ref.WeakReference
 
 class RecreateReactContextProcedure(
-  private val reactNativeHost: WeakReference<ReactNativeHost>?,
+  private val context: Context,
+  private val weakActivity: WeakReference<Activity>?,
   private val callback: Launcher.LauncherCallback
 ) : StateMachineProcedure() {
   override val loggerTimerLabel = "timer-recreate-react-context"
 
   override fun run(procedureContext: ProcedureContext) {
-    val host = reactNativeHost?.get()
-    if (host == null) {
+    val reactApplication = context.applicationContext as? ReactApplication ?: run {
```

Comment:
prefer to give this block a different return name than the outer function named `run` just to be sure the `return` within the block isn't just for the block.

```suggestion
    val reactApplication = context.applicationContext as? ReactApplication ?: run inner@ {
```